modules/widgets/simple-payments/customizer.js

```diff
@@ -4,6 +4,15 @@
 ( function( api, wp, $ ) {
 	var $document = $( document );
 
+	// based on https://stackoverflow.com/a/10454560/59752
+	function decimalPlaces( number ) {
+		var match = ( '' + number ).match( /(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/ );
+		if ( ! match ) {
+			return 0;
+		}
+		return Math.max( 0, ( match[ 1 ] ? match[ 1 ].length : 0 ) - ( match[ 2 ] ? +match[ 2 ] : 0 ) );
```

Comment:
not sure we really need to handle scientific notation here. we can probably validate our currency input where incomes in then use something much simpler and easer to reason about in order to count digits.

```js
// linked code is still based on decimal numbers instead of currency values
// so this still assumes a `.` where in reality we may have a `,`
function givenDecimalPlaces( currencyString ) {
	return currencyString.replace( /[^.]+./, '' ).length;

	// or
	var fractional = currencyString.split( '.' )[ 1 ];
	return fractional ? fractional.length : 0;

	// or again even more specific with locale
	var match = currencyString.match( /^(?:(?:\d{1,3},)*\d{3}|(?:\d+))(?:(?:\.)(\d+))$/ );
	return match ? match[ 1 ].length : 0;
}
```